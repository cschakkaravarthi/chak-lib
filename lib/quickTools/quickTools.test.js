"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var testHelper_1 = require("../services/testHelper");
var quickToolsCard_1 = require("./quickToolsCard");
describe('QuickTools', function () {
    it('should render correctly', function () {
        var tree = testHelper_1.shallowRender(react_1["default"].createElement(quickToolsCard_1.BasicQuickToolsCard, { title: "Something's Broken!", Links: [
                {
                    linkAction: function (p) { return react_1["default"].createElement("a", { href: "#" }, p.children); },
                    linkTitle: 'Create an i.t. ticket'
                }
            ] }));
        expect(tree).toMatchSnapshot();
    });
    it('should render correctly', function () {
        var tree = testHelper_1.shallowRender(react_1["default"].createElement(quickToolsCard_1.BasicQuickToolsCard, { title: "Self-Service Password Reset", Links: [
                {
                    linkAction: function (p) { return react_1["default"].createElement("a", { href: "#" }, p.children); },
                    linkTitle: 'Reset Password'
                }
            ] }));
        expect(tree).toMatchSnapshot();
    });
    it('should render correctly', function () {
        var tree = testHelper_1.shallowRender(react_1["default"].createElement(quickToolsCard_1.BasicQuickToolsCard, { title: "Time off request", Links: [
                {
                    linkAction: function (p) { return react_1["default"].createElement("a", { href: "#" }, p.children); },
                    linkTitle: 'Request time off'
                }
            ] }));
        expect(tree).toMatchSnapshot();
    });
    it('should render correctly', function () {
        var tree = testHelper_1.shallowRender(react_1["default"].createElement(quickToolsCard_1.BasicQuickToolsCard, { title: "Important Contacts", Links: [
                {
                    linkAction: function (p) { return react_1["default"].createElement("a", { href: "#" }, p.children); },
                    linkTitle: 'Security'
                },
                {
                    linkAction: function (p) { return react_1["default"].createElement("a", { href: "#" }, p.children); },
                    linkTitle: 'Facilities'
                }
            ] }));
        expect(tree).toMatchSnapshot();
    });
});
//# sourceMappingURL=quickTools.test.js.map