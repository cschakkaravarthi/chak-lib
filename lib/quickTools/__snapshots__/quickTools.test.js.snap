// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickTools should render correctly 1`] = `
<ForwardRef
  body={false}
  className="py-4 px-3 flex-column justify-content-between h-100 rounded-0"
>
  <ForwardRef
    className="font-weight-bold text-capitalize text-condensed font-size-base mb-4"
  >
    Something's Broken!
  </ForwardRef>
  <linkAction>
    <u
      className="text-upper gray-600 font-size-sm letter-spacing"
    >
      Create an i.t. ticket
    </u>
  </linkAction>
</ForwardRef>
`;

exports[`QuickTools should render correctly 2`] = `
<ForwardRef
  body={false}
  className="py-4 px-3 flex-column justify-content-between h-100 rounded-0"
>
  <ForwardRef
    className="font-weight-bold text-capitalize text-condensed font-size-base mb-4"
  >
    Self-Service Password Reset
  </ForwardRef>
  <linkAction>
    <u
      className="text-upper gray-600 font-size-sm letter-spacing"
    >
      Reset Password
    </u>
  </linkAction>
</ForwardRef>
`;

exports[`QuickTools should render correctly 3`] = `
<ForwardRef
  body={false}
  className="py-4 px-3 flex-column justify-content-between h-100 rounded-0"
>
  <ForwardRef
    className="font-weight-bold text-capitalize text-condensed font-size-base mb-4"
  >
    Time off request
  </ForwardRef>
  <linkAction>
    <u
      className="text-upper gray-600 font-size-sm letter-spacing"
    >
      Request time off
    </u>
  </linkAction>
</ForwardRef>
`;

exports[`QuickTools should render correctly 4`] = `
<ForwardRef
  body={false}
  className="py-4 px-3 flex-column justify-content-between h-100 rounded-0"
>
  <ForwardRef
    className="font-weight-bold text-capitalize text-condensed font-size-base mb-4"
  >
    Important Contacts
  </ForwardRef>
  <linkAction>
    <u
      className="text-upper gray-600 font-size-sm letter-spacing"
    >
      Security
    </u>
  </linkAction>
  <linkAction>
    <u
      className="text-upper gray-600 font-size-sm letter-spacing"
    >
      Facilities
    </u>
  </linkAction>
</ForwardRef>
`;
